pkg/core/project_store.go

```diff
@@ -0,0 +1,48 @@
+package core
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+
+	"github.com/fuseml/fuseml-core/pkg/domain"
+)
+
+// GitProjectStore describes a stucture that accesses project store implemented in git
+type GitProjectStore struct {
```

Comment:
I do not think it as so different than other artifacts that it would require different treatment...

Manager might be better but maybe I'll just refactor it again after we have manager implemented for workflows